incubator/orm/scenario_test.go

```diff
@@ -0,0 +1,263 @@
+	return &MockContext{
+		db:    dbm.NewMemDB(),
+		store: store.NewCommitMultiStore(db),
+	}
+
+}
+func (m MockContext) KVStore(key sdk.StoreKey) sdk.KVStore {
+	if s := m.store.GetCommitKVStore(key); s != nil {
+		return s
+	}
+	m.store.MountStoreWithDB(key, sdk.StoreTypeIAVL, m.db)
+	if err := m.store.LoadLatestVersion(); err != nil {
+		panic(err)
+	}
+	return m.store.GetCommitKVStore(key)
+}
+
+func TestKeeperEndToEndWithAutoUInt64Table(t *testing.T) {
+	storeKey := sdk.NewKVStoreKey("test")
+	cdc := codec.New()
+	ctx := NewMockContext()
+
+	k := NewGroupKeeper(storeKey, cdc)
+
+	g := GroupMetadata{
+		Description: "my test",
+		Admin:       sdk.AccAddress([]byte("admin-address")),
+	}
+	// when stored
+	rowID, err := k.groupTable.Create(ctx, &g)
+	if err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	// then we should find it
+	exists, _ := k.groupTable.Has(ctx, rowID)
+	if exp, got := true, exists; exp != got {
+		t.Fatalf("expected %v but got %v", exp, got)
+	}
+	// and load it
+	it, err := k.groupTable.Get(ctx, rowID)
+	if err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	binKey, loaded := first(t, it)
+	if exp, got := rowID, binary.BigEndian.Uint64(binKey); exp != got {
+		t.Errorf("expected %v but got %v", exp, got)
+	}
+	if exp, got := "my test", loaded.Description; exp != got {
+		t.Errorf("expected %v but got %v", exp, got)
+	}
+	if exp, got := sdk.AccAddress([]byte("admin-address")), loaded.Admin; !bytes.Equal(exp, got) {
+		t.Errorf("expected %X but got %X", exp, got)
+	}
+	// and exists in index
+	exists, err = k.groupByAdminIndex.Has(ctx, []byte("admin-address"))
+	if err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	if !exists {
+		t.Fatalf("expected entry to exist")
+	}
+	// and when loaded
+	it, err = k.groupByAdminIndex.Get(ctx, []byte("admin-address"))
+	if err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	// then
+	binKey, loaded = first(t, it)
+	if exp, got := rowID, binary.BigEndian.Uint64(binKey); exp != got {
+		t.Errorf("expected %v but got %v", exp, got)
+	}
+	if exp, got := g, loaded; !reflect.DeepEqual(exp, got) {
+		t.Errorf("expected %v but got %v", exp, got)
+	}
+	// when updated
+	g.Admin = []byte("new-admin-address")
+	err = k.groupTable.Save(ctx, rowID, &g)
+	if err != nil {
+		t.Fatalf("unexpected error: %+v", err)
+	}
+	// then indexes are updated, too
+	exists, _ = k.groupByAdminIndex.Has(ctx, []byte("new-admin-address"))
+	if exp, got := true, exists; exp != got {
+		t.Fatalf("expected %v but got %v", exp, got)
+	}
+	exists, _ = k.groupByAdminIndex.Has(ctx, []byte("admin-address"))
+	if exp, got := false, exists; exp != got {
+		t.Fatalf("expected %v but got %v", exp, got)
+	}
+
+	// when deleted
+	k.groupTable.Delete(ctx, rowID)
+
+	// then removed from primary index
+	exists, _ = k.groupTable.Has(ctx, rowID)
+	if exp, got := false, exists; exp != got {
+		t.Fatalf("expected %v but got %v", exp, got)
+	}
+	// and also removed from secondary index
```

Comment:
ja, I will break down the scenario tests into proper integration tests once the api is stable.

As a note: the secondary index key contains the admin address and the rowID. The 3 groups would result in 3 entries (with different keys) on the second index. They are not grouped by the admin key. In case of an unique index a prefix scan makes sure that only 1 entry exists for an admin address.